src/alexaSkillDemo.py

```diff
@@ -0,0 +1,235 @@
+                    'You can tell me your building by saying, my building is.'
+    
+    """If the user either does not reply to the welcome message or says something
+       that is not understood, they will be prompted again with this text.
+    """
+    repromptText = 'Please tell me your building.' \
+                   'You can tell me your building by saying, my building is.'
+    shouldEndSession = False
+    return buildResponse(sessionAttributes, buildSpeechletResponse(
+        cardTitle, speechOutput, repromptText, shouldEndSession))
+
+def handleSessionEndRequest():
+    cardTitle = 'Session Ended'
+    speechOutput = 'Thank you for trying the Why Not Change Your Clothes? Alexa skill demo.' \
+                    'Have a nice day! '
+    """Setting this to true ends the session and exits the skill."""
+    shouldEndSession = True
+    return buildResponse({}, buildSpeechletResponse(
+        cardTitle, speechOutput, None, shouldEndSession))
+
+def createBuildingAttributes(building):
+    return {'building': building}
+
+def setBuildingInSession(intent, session):
+    """ Sets the building in the session and prepares the speech to reply to the user."""
+    cardTitle = intent.get('name')
+    sessionAttributes = {}
+    shouldEndSession = False
+
+    if 'building' in intent.get('slots'):
+        building = intent.get('slots').get('building').get('value')
+        sessionAttributes = createBuildingAttributes(building)
+        speechOutput = 'I now know your building is ' + building.capitalize() + '. ' \
+                       'You can ask me what your building is by saying, what is my building?'
+        repromptText = 'You can ask me what your building is by saying, what is my building?'
+        
+        """Calls /buildings?friendlyName= API Call to find valid buildingId"""
+        findBuilding = callApi(
+            "https://go3ba09va5.execute-api.us-east-2.amazonaws.com/Test/buildings?friendlyName={}".format(building)
+        )
+        
+        if findBuilding.get('buildingId'):
+            user = dict()
+            user['userId'] = session.get('user').get('userId')
+            user['buildingId'] = findBuilding.get('buildingId')
+            writeDynamo(user)
+        else:
+            speechOutput = 'I could not find that building. Please try again.'\
+                           'You can tell me your building by saying, my building is.'
+            repromptText = 'You can tell me your building by saying, my building is.'
+        
+    else:
+        speechOutput = 'I am not sure what your building is. Please try again.'
+        repromptText = 'I am not sure what your building is. Please try again.' \
+                       'You can tell me your building by saying, my building is.'
+    return buildResponse(sessionAttributes, buildSpeechletResponse(
+        cardTitle, speechOutput, repromptText, shouldEndSession))
+
+def getBuildingFromSession(intent, session):
```

Comment:
Make sure to read from Dynamo later! In most cases you won't be in the same session as when it's set, so Dynamo is going to serve as our "source of truth" essentially. In fact, you can probably scrap working with the session at all, unless we do something later that requires state between calls (that we wouldn't put in Dynamo for some reason).